src/env.h

```diff
@@ -1384,9 +1371,7 @@ class Environment : public MemoryRetainer {
   bool http_parser_buffer_in_use_ = false;
   std::unique_ptr<http2::Http2State> http2_state_;
 
-  bool debug_enabled_[static_cast<int>(DebugCategory::CATEGORY_COUNT)] = {
-      false};
-
+  std::unique_ptr<EnabledDebugList> enabled_debug_list_;
```

Comment:
Is there any reason that this would need to be a pointer? I don’t think that’s necessary, and it comes with one additional layer of indirection for each debug check. One of the goals for the Debug() implementation was to have as close to zero overhead as possible, and I think this might get in the way of it…

Reply:
Mostly to avoid the circular dependency - but I guess this can also be achieved by moving more things from `debug_utils.h` to `debug_utils-inl.h` so that `env.h` can just include `debug_utils.h` and `debug_utils.h` do not need to include `env.h` ?

Reply:
If that works, then that would be great – otherwise we might need to move more things into `env.h`…

Reply:
hmm, looks like if we want to keep the initialization in the constructor, then it needs to be a pointer since we'd need the environment pointer with initialized env_vars for `credentials::SafeGetenv()`. So I just moved the initialization into `EnableDebugList::Parse()`, and call it in `InitializeOncePerProcess` (and separately, in mkcodecache since it does not need `InitializeOncePerProcess()`), this also means that for now the printer is not usable until `node::Start()` is called in the default binary, but I guess that's fine, compared to reordering the whole initialization sequence to work around pointers.

Reply:
Fwiw, what I had in mind is to use the trivial copy constructor to re-initialize it later, e.g. `enabled_debug_list_ = EnabledDebugList(...);`, but I guess `::Parse()` works just as well :)